tests/automated_tests/test_provider_and_devices.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2023 University of Stuttgart
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""""Test class to test the functionality of the job_api"""
+from qunicorn_core.api.api_models import ProviderDto, DeviceDto, SimpleDeviceDto
+from qunicorn_core.core import provider_service, device_service
+from tests.conftest import set_up_env
+
+
+def test_get_all_provider():
+    """Testing if the get all endpoint of providers works"""
```

Comment:
Again as with 'test get all deployments' or maybe just make 'endpoint' plural (?) - I don't really understand the comment